src/Data/Array/Accelerate/Smart.hs

```diff
@@ -215,8 +215,8 @@ import GHC.TypeLits
 --
 -- > mvm :: A.Num a => Acc (Matrix a) -> Acc (Vector a) -> Acc (Vector a)
 -- > mvm mat vec =
--- >   let Z :. rows :. cols = unlift (shape mat) :: Z :. Exp Int :. Exp Int
--- >       vec'              = A.replicate (lift (Z :. rows :. All)) vec
+-- >   let I2 rows cols = shape mat
+-- >       vec'         = A.replicate (rows ::. All_) vec
 -- >   in
 -- >   A.fold (+) 0 ( A.zipWith (*) mat vec' )
```

Comment:
... why is this not a type error? `rows :: Exp Int`, so that `::.` makes no sense.

Why not:
```
mvm :: A.Num a => Acc (Matrix a) -> Acc (Vector a) -> Acc (Vector a)
mvm mat vec =
  let I2 rows cols = shape mat
      vec'         = A.replicate (I2 rows All_) vec
  in
  A.sum (A.zipWith (*) mat vec')
```

Note: I also removed spaces inside the `( )` and replaced the explicit `fold` with a `sum`, which it is.

Reply:
This is exactly the type of error I hoped the doctest would catch, even if it did not execute.. I guess it doesn't run on these parts? I should read up on what it does exactly...
Lesson learned, this time I'm actually testing the code first!

Reply:
Yeah I have no idea. :)